build-tools/packages/version-tools/test/semver.test.ts

```diff
@@ -78,6 +78,54 @@ describe("semver", () => {
         });
     });
 
+    describe("detectBumpType internal version scheme", () => {
+        it("major", () => {
+            assert.equal(detectBumpType("2.0.0-internal.1.0.0", "2.0.0-internal.2.0.0"), "major");
+        });
+
+        it("minor", () => {
+            assert.equal(detectBumpType("2.0.0-internal.1.0.0", "2.0.0-internal.1.1.0"), "minor");
+        });
+
+        it("patch", () => {
+            assert.equal(detectBumpType("2.0.0-internal.1.0.0", "2.0.0-internal.1.0.1"), "patch");
+        });
+
+        it("premajor", () => {
+            assert.equal(
+                detectBumpType("2.0.0-internal.1.0.0.82134", "2.0.0-internal.2.0.0"),
+                "major",
+            );
+        });
+
+        it("preminor", () => {
+            assert.equal(
+                detectBumpType("2.0.0-internal.1.1.0.82134", "2.0.0-internal.1.2.0"),
+                "minor",
+            );
+        });
+
+        it("prepatch", () => {
+            assert.equal(detectBumpType("1.1.1-foo", "1.1.2"), "patch");
+        });
+
+        it("prerelease", () => {
+            assert.isUndefined(
+                detectBumpType("2.0.0-internal.1.0.0.82134", "2.0.0-internal.1.0.0"),
+            );
+        });
+
+        it("v1 >= v2 throws", () => {
+            assert.throws(() => detectBumpType("0.0.1", "0.0.1"));
+            assert.throws(() => detectBumpType("0.0.2", "0.0.1"));
+            assert.throws(() => detectBumpType("0.0.1+0", "0.0.1"));
+            assert.throws(() => detectBumpType("0.0.1+2", "0.0.1+2"));
```

Comment:
Is this valid semver (0.0.1+2)? If not, maybe you want to validate against bad semver in general.

Reply:
It is valid in that it is parseable by the semver lib. I added test cases with known bad semver values for both arguments.